tests/kernel/mem_protect/mem_protect/src/mem_domain.c

```diff
@@ -130,8 +147,88 @@ void test_mem_domain_invalid_access(void *p1, void *p2, void *p3)
 
 	k_sem_take(&sync_sem, SYNC_SEM_TIMEOUT);
 }
+/***************************************************************************/
+static void thread_entry_rw(void *p1, void *p2, void *p3)
+{
+	u8_t read_data = mem_domain_buf[0];
+
+	/* Write to the partition */
+	mem_domain_buf[0] = 99;
 
+	zassert_false(read_data == mem_domain_buf[0], "Read and write to"
+			" partition has failed\n");
+
+	/* Reset the value */
+	mem_domain_buf[0] = 0;
+
+	ztest_test_pass();
+}
+/* Provide read/write access to a partition and verify access from a
+ * user thread added to it
+ */
+void test_mem_domain_partitions_user_rw(void)
+{
+	/* Initialize the memory domain */
+	k_mem_domain_init(&mem_domain_mem_domain,
+			MEM_PARTITION_INIT_NUM,
+			mem_domain_memory_partition_array);
+
+	k_mem_domain_add_thread(&mem_domain_mem_domain,
+			k_current_get());
+
+	k_thread_user_mode_enter(thread_entry_rw, NULL, NULL, NULL);
+}
 /****************************************************************************/
+static void user_thread_entry_ro(void *p1, void *p2, void *p3)
+{
+	valid_fault = true;
+
+	printk("Read the partition: %d\n", mem_domain_buf1[0]);
+	USERSPACE_BARRIER;
+
+	/* Writing to the partition, this should generate fault
+	* as the partition has read only permission for
+	* user threads
+	*/
+    mem_domain_buf1[0] = 10;
+
+	zassert_unreachable("The user thread is allowed to access a read only"
+			" partition of a memory domain\n");
+}
+
+void test_mem_domain_partitions_user_ro(void)
+{
+	/* Initialize the memory domain containing the partition
+	 * with read only access priviledge
+	 */
+	k_mem_domain_init(&mem_domain1,
+			MEM_PARTITION_INIT_NUM,
+			mem_domain_memory_partition_array1);
+
+	k_mem_domain_add_thread(&mem_domain1, k_current_get());
+
+	k_thread_user_mode_enter(user_thread_entry_ro, NULL, NULL, NULL);
+}
+
+/****************************************************************************/
+void test_mem_domain_partitions_supervisor_rw(void)
```

Comment:
I am not totally sure if this test case is needed. Because in SU we will be able to access everything and that has been tested already

Reply:
IMHO, having a explicit test would be good so that it would be easy to identify any faults in read/write operations on partition in future.